pkg/logging/logging.go

```diff
@@ -235,20 +245,32 @@ func setupSyslog(logOpts map[string]string, tag string, debug bool) {
 	logrus.AddHook(h)
 }
 
-// setupFormatter sets up the text formatting for logs output by logrus.
-func setupFormatter() logrus.Formatter {
-	fileFormat := new(logrus.TextFormatter)
-	fileFormat.DisableTimestamp = true
-	fileFormat.DisableColors = true
-	switch os.Getenv("INITSYSTEM") {
-	case "SYSTEMD":
-		fileFormat.FullTimestamp = true
-	default:
-		fileFormat.TimestampFormat = time.RFC3339
+// GetFormatter returns a configured logrus.Formatter with some specific values
+// we want to have
+func GetFormatter(kind string) (logrus.Formatter, bool) {
+	switch kind {
+	case "text":
+		formatter := &logrus.TextFormatter{
+			DisableTimestamp: true,
+			DisableColors:    true,
+		}
+
+		// Question for reviewers: Is this actually useful given that timestamps marshalling is always
+		// disabled?
```

Comment:
Looks like the timestamp disabling didn't take into account the rest of these changes, so no I don't see how this makes sense, seems to me like we should be able to drop these timestamp formatting bits.

Also, we would normally separate out independent changes like these logging changes to at least a separate commit, if not even a separate PR. I don't see how they relate to the goal of this PR, and without explanation I'm not sure what problem you're trying to solve here.

They seem to be abstracting out the usage for supporting JSON logging output formatters, which sounds like a feature that maybe is useful to others but should be called out if we want people to use it. Easiest way to do so is to file a separate PR, then when we release the next version of Cilium with this PR, we will pick up the PR title and list it in the release notes and folks can find it from there. (I also don't see any JSON users of the new functions so I don't understand how to validate these changes, but maybe you have something in mind there...?)

Reply:
apologies, this file got out of my git stash 🤦 it was totally meant for another PR..! I removed it 😅

Reply:
I moved it in a draft as I was not done yet : https://github.com/cilium/cilium/pull/11133
Thanks a lot for the insights on my comment @joestringer!